org.metaborg.spoofax.shell.eclipse/src/main/java/org/metaborg/spoofax/shell/client/eclipse/impl/EclipseRepl.java

```diff
@@ -85,11 +88,10 @@ private void runAsJob(final String input) {
             @Override
             protected IStatus run(IProgressMonitor monitor) {
                 try {
-                    runAsUIJob(eval(input));
+                    IResult result = pool.submit(() -> eval(input)).get();
+                    runAsUIJob(result);
                     return Status.OK_STATUS;
-                } catch (CommandNotFoundException e) {
-                    StyledText message = new StyledText(Color.RED, e.getMessage());
-                    runAsUIJob((visitor) -> visitor.visitMessage(message));
+                } catch (InterruptedException | ExecutionException e) {
                     return Status.CANCEL_STATUS;
```

Comment:
To close the REPL programmatically, use:

```
IWorkbenchPage page = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage();
IViewPart replView = page.findView(ReplView.ID);
page.hideView(replView);
```

To have this work, you need to add the following to:

* `Activator`:
```
public static final String ID = "org.metaborg.spoofax.shell.eclipse";
```
* `ReplView`:
```
public static final String ID = Activator.ID + ".view";